R/database.R

```diff
@@ -2628,3 +2636,258 @@ get_gene_expression = function(metadata,
+#' @param min_score The lowest Manta somatic score for a SV to be returned. Default value is 40.
+#' @param pass If set to TRUE, include SVs that are annotated with PASS in FILTER column. Default is TRUE.
+#' @param with_chr_prefix Toggle if chromosome names should be prefixed or not, default is FALSE.
+#' @param projection The projection of returned calls. Default is grch37.
+#' 
+#' @return a data frame containing the Manta outputs from all sample_id in these_samples_metadata in a bedpe-like format with additional columns extracted from the VCF column.
+#' @import tidyverse
+#' @export
+#'
+#' @examples
+#' all_sv = get_manta_sv
+#' missing_samples = get_gambl_metadata() %>% anti_join(all_sv, by = c("sample_id" = "tumour_sample_id"))
+#' missing_from_merge = get_manta_sv_by_samples(these_samples_metadata = missing_samples)
+#'
+get_manta_sv_by_samples = function(these_samples_metadata,
+                                   min_vaf = 0.1,
+                                   min_score = 40,
+                                   pass = TRUE,
+                                   with_chr_prefix = FALSE,
+                                   projection = "grch37"){
+  
+  #check remote configuration
+  remote_session = check_remote_configuration(auto_connect = TRUE)
+
+  #get sample IDs from metadata.
+  samples = pull(these_samples_metadata, sample_id)
+  
+  #create an empty list.
+  all_bedpe = list()
+  
+  #wrap get_manta_sv_by_sample.
+  all_bedpe = lapply(samples, function(x){get_manta_sv_by_sample(this_sample_id = x,
+                                                                 these_samples_metadata = these_samples_metadata,
+                                                                 force_lift = FALSE,
+                                                                 min_vaf = min_vaf,
+                                                                 min_score = min_score,
+                                                                 pass = pass,
+                                                                 with_chr_prefix = with_chr_prefix,
+                                                                 projection = projection)})
+  
+  #un-nest list into long format.
+  merged_bedpe = bind_rows(all_bedpe)
+  
+  #take out all calls that need to be lifted
+  to_be_lifted = merged_bedpe %>%
+    dplyr::filter(need_lift == TRUE)
+  
+  #lift to selected projection
+  lifted_calls = liftover_bedpe(bedpe_df = to_be_lifted, target_build = projection, print_head = FALSE)
+  
+  #subset calls that does not need a "lift"
+  no_lift_needed = merged_bedpe %>%
+    dplyr::filter(need_lift == FALSE)
+  
+  #combine calls (lifted and not lifted), arrange and sort accordingly, drop temporary column
+  merged_bedpe = rbind(lifted_calls, no_lift_needed) %>%
+    arrange(CHROM_A, CHROM_B, START_A) %>%
+    dplyr::select(-need_lift)
+  
+  #return merged manta SVs.
```

Comment:
Not sure why there are rownames that need to be dropped. Is this because you used rbind instead of bind_rows maybe?

Reply:
The row names were dropped since my approach for dealing with imprecise variants scrambled all rows when stitching the data back together. It has no effect on the data frame, I thought it looked cleaner not to have row names in this data frame. However, `bind_rows()` solves this without setting row names to NULL. Both `get_manta_by_x` functions have been updated to use `bind_rows` instead, good suggestion!